lib/services/tendermintCore/TendermintCore.js

```diff
@@ -0,0 +1,140 @@
+const DockerService = require('../../docker/DockerService');
+const wait = require('../../util/wait');
+
+class TendermintCore extends DockerService {
+  /**
+   * Create TendermintCore instance
+   *
+   * @param {Network} network
+   * @param {Image} image
+   * @param {Container} container
+   * @param {TendermintClient} TendermintClient
+   * @param {TendermintCoreOptions} options
+   */
+  constructor(network, image, container, TendermintClient, options) {
+    super(network, image, container, options);
+    this.TendermintClient = TendermintClient;
+    this.options = options;
+  }
+
+  /**
+   * Start instance
+   *
+   * @return {Promise<void>}
+   */
+  async start() {
+    await this.network.create();
+    await this.image.pull();
+    await this.initTendermintGenesisBlock();
+    await super.start();
+    await this.initialize();
+  }
+
+  async stop() {
+    await this.tendermintClient.close();
+
+    await super.stop();
+  }
+
+  /**
+   * Clean blockchain
+   *
+   * @returns {Promise}
+   */
+  async clean() {
+    await this.remove();
+    await this.start();
+  }
+
+  async remove() {
+    await this.tendermintClient.close();
+    await super.remove();
+
+    const volume = await this.container.docker.getVolume(this.options.getTendermintVolumeName());
+    await volume.remove();
+  }
+
+  /**
+   * @private
+   *
+   * @return {Promise<void>}
+   */
+  async initialize() {
+    let nodeStarting = true;
+    while (nodeStarting) {
+      try {
+        const tendermintClient = this.createTendermintClient('http');
+        await tendermintClient.status();
+
+        nodeStarting = false;
+        await this.tendermintClient.close();
+      } catch (error) {
+        await wait(100);
```

Comment:
Don't we want to check a specific error? It might be everything and then we run into an infinite loop.